### ISSUE TITLE
Bodymovin converts source images to large PNG files. Why?

### ISSUE THREAD

**Issue by jasonguerra** [airbnb/lottie-web#1492]
When using images in a AE comp, Bodymovin exports the images as PNG files regardless of whether the source is a PNG or JPG, and increases the image byte size considerably.

With "enable compression" turned OFF in the Bodymovin settings:
- PNG is exported at a byte size that is larger than the source PNG.
- JPG is converted to PNG and exported at a byte size much larger than the source JPG.

The only way to get a source JPG to export as a JPG is to enable compression, which may not always be desired.

Ideally, Bodymovin should just pass source images straight through without any format conversion or expansion/compression (unless compression is enabled). That way animators can set the desired image type and compression level of assets before they are used in a comp. Image byte sizes should not be getting bigger. JPGs should not be getting converted to PNGs.

This issue also affects the "include in JSON" option. The source image file size gets inflated before getting encoded into the JSON, which makes the JSON unnecessarily large.

**Comment by bodymovin** [comment]
Hi, regarding image sources in general, there is room for improvement indeed.
But it's difficult to identify which assets are originally PNGs, JPGs, ai layers, photoshop layers, or other types of formats reliably.
The decision of using PNG as the default format is because of it's lossless nature that can guarantee that images don't lose quality.
If the original file is a JPG, compressing it will always lose quality, but you can decide amount of compression in the UI to prevent degrading them even more.
You can always replace the images folder with your own files to make sure you get the desired results.

**Comment by jasonguerra** [comment]
Thanks, I did figure out that I could just replace the assets in the images folder with my own optimized files after rendering. But I'm most interested in using the "include in JSON" feature to encapsulate everything into one file. Unfortunately there's no way to optimize images once they get combined with the JSON. So when the image byte size gets enlarged during rendering by Bodymovin, that's the version that get wrapped into JSON.

**Comment by tinogithub** [comment]
Hi,
I came across the same problem. Unfortunately it is not just replacing the images folder because the filenames in the html and json change from jpg to png after conversion. So to really use the source images you need to change the source manually. Moreover the script only converts to PNG sometimes. 
I have two JPEGs in my comp and the one is being converted and the other not.
Why is that?

**Comment by bodymovin** [comment]
@tinogithub the script only converts to jpgs if those layers have no transparency

**Comment by tinogithub** [comment]
@bodymovin I tried it. I created a new comp and just inserted a jpeg picture which of course has no transparency. Did nothing to it else to it. When I export it, the script converts to PNG. How does the the script think that there is transparency?
thank you for your help and by the way ... despite these little problems: you did some great work with it. Thank you. :-)

**Comment by bodymovin** [comment]
can you share the .aep so I can take a look?

**Comment by tinogithub** [comment]
Sure,
here you are.
[JPG-PNG Ordner.zip](https://github.com/airbnb/lottie-web/files/2962896/JPG-PNG.Ordner.zip)

thank you.

**Comment by bodymovin** [comment]
I guess I didn't explain it fully, sorry.
Before exporting you need to enable compression on the bodymovin composition settings.
Select Assets > Enable compression and choose the level of compression of the image you want.

**Comment by tinogithub** [comment]
Hi,
thank you for this. However I still don't understand the method behind it.
Now I've tried your explanation. However, now it converts from jpg to png and vice versa.
![grafik](https://user-images.githubusercontent.com/23121904/55064655-a66e7600-507a-11e9-9940-d648fe8faaf8.png)
Here you see some files. On the left my original footage and on the right what bodymovin is making out of it during export. I enabled compression with 80%.
Especially the big PNG is strange. Why does it do this?

Here is the project:
[jpg_png_test_2.zip](https://github.com/airbnb/lottie-web/files/3012442/jpg_png_test_2.zip)

thank you for your help,

**Comment by bodymovin** [comment]
In order to compress an image, the library draws the exported image to a canvas, then clones that canvas with a white background and compares the original image with the cloned one.
If there is one pixel that is different between them, it doesn't compress the image because it assumes it has at least one transparent pixel.
I don't know why, but for `2007_05_30_glo_007.JPG`, when it compares it, it finds several pixels that are different (around 60 pixels total), and that's why it defaults to png. Perhaps because of size of the image
 but I'm not sure. The other 3 images, even with the .png one, since it doesn't find any pixels that are different, it converts all of them to jpgs.

**Comment by bodymovin** [comment]
I've tried a different approach to compare original and cloned images and it seems to work better, so I'll publish it on the next deploy.
Thanks for reporting it.

**Comment by tinogithub** [comment]
This is good news.
Thank you very much. I'll give it a try.

**Comment by bodymovin** [comment]
@materia-admin have you tried selecting use original assets?

**Comment by desidem** [comment]
Hi, I'm experiencing something similar. I am using PNG files in my After Effects project (Bodymovin extension version 5.7.4). 

I started out with PNG-24 (created in Illustrator) but when I saw the export size for the JSON files (many in the neighborhood of 8MB), I decided to export as PNG-8 128 dithered for a reduced file PNG. 

My PNG sizes reduced significantly (e.g., 2.7MB to 615KB).

<img width="1061" alt="Screen Shot 2022-07-03 at 8 21 30 PM" src="https://user-images.githubusercontent.com/29647018/177062938-1ca78557-e6df-4d5e-8206-71c003223103.png">

In After Effects, I replaced the larger PNG-24 files with new, reduced files (PNG-8). 
<img width="482" alt="Screen Shot 2022-07-03 at 8 22 08 PM" src="https://user-images.githubusercontent.com/29647018/177063043-c1f15d70-6417-4f7d-a967-94c249131da5.png">

But after using the Bodymovin/Lottie extension, the resulting JSON file was still huge (7.1 MB). I selected for compression under the Assets setting too. 

Any advice would be much appreciated. I'm trying to get the file sizes down. I thought re-exporting the files from Adobe Illustrator as PNG-8 128 would solve it. 

Thanks.
